server/src/api/post.rs

```diff
@@ -100,6 +103,15 @@ impl Perform<PostResponse> for Oper<CreatePost> {
 
     let user_id = claims.id;
 
+    if let Some(rl) = &rate_limit_info {
+      rl.rate_limiter
+        .lock()
+        .unwrap()
+        .check_rate_limit_post(&rl.ip, true)?;
```

Comment:
What's the reasoning around not updating the allowance for PostResponse, but updating it for literally anything else?

Reply:
You mean the check_only = true?

So right now I'm handling post, community, and registers in a hacky way in order to take care of errors.

First it checks the rate limit, but doesn't do a penalty for checking. Then it tries to create the post, and if it goes through, it does the penalty at the end of the function.

I'm doing it this way so a post failure doesn't incur a penalty.

Reply:
Sounds a bit like premature optimisation.

Reply:
Is that functionality that you'd like for all requests (i.e. only penalizing for successful operations), or is creating a post special?

Reply:
I'm honestly not sure the correct way to do it, since errors (things like slurs in the post, or similar usernames, etc) could be thrown at any time there. And I obviously have to do a check first otherwise every action would go through.

Reply:
@asonix it only does that double check for 3 things: account creation, community creation (these both use `check_rate_limit_register), and post creation. All the other actions use the `check_rate_limit_message` one which is so high that it really shouldn't matter.